src/ui_utils.c

```diff
@@ -2440,20 +2441,32 @@ void ui_init_builder(void)
 	gtk_builder_set_translation_domain(builder, GETTEXT_PACKAGE);
 
 	error = NULL;
-	interface_file = g_build_filename(app->datadir, "geany.glade", NULL);
-	if (! gtk_builder_add_from_file(builder, interface_file, &error))
+	ui_data = g_resource_lookup_data(geany_get_resource(),
+		"/org/geany/Geany/geany.glade", G_RESOURCE_LOOKUP_FLAGS_NONE, &error);
+	if (ui_data == NULL)
```

Comment:
what about combining this with the check below with a `||`, as the body is almost totally the same (but for the `ui_data` unrefing part)

Reply:
Could do, I just wanted to show two discrete messages (ie. "load" vs "create"). I don't mind merging them, both are super-fatal.

Reply:
Oh, I didn't notice that.  I don't think it matters much though.